webdriver/newsession/invalid_capabilities.py

```diff
@@ -0,0 +1,88 @@
+#META: timeout=long
+
+import pytest
+from webdriver import error
+
+from conftest import product, flatten
+
+
+@pytest.mark.parametrize("value", [None, 1, "{}", []])
+def test_invalid_capabilites(new_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({"capabilities": value})
+
+
+@pytest.mark.parametrize("value", [None, 1, "{}", []])
+def test_invalid_always_match(new_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({"capabilities": {"alwaysMatch": value}})
+
+
+@pytest.mark.parametrize("value", [None, 1, "[]", {}])
+def test_invalid_first_match(new_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session({"capabilities": {"firstMatch": value}})
```

Comment:
What? 

```
Let all first match capabilities be the result of getting the property "firstMatch" from capabilities request.

    If all first match capabilities is undefined, set the value to a JSON List with a single entry of an empty JSON Object.

    If all first match capabilities is not a JSON List, return error with error code invalid argument. 
```

If it's not a list it's clearly invalid, and I don't think that should change.